src/main/java/teammates/ui/controller/AdminActivityLogPageAction.java

```diff
@@ -259,16 +258,13 @@ private void generateStatusMessage(List<String> versionToQuery,
             }
             
             ActivityLogEntry activityLogEntry = new ActivityLogEntry(appLog);
-            activityLogEntry = data.filterLogs(activityLogEntry);
             
-            boolean isToShow = activityLogEntry.toShow() && (!activityLogEntry.isTestingData() || data.getIfShowTestData());
+            boolean isToShow = data.filterLogs(activityLogEntry)
+                                && (!activityLogEntry.isTestingData() || data.getIfShowTestData());
```

Comment:
`(!activityLogEntry.isTestingData() || data.getIfShowTestData())` can be possibly merged into the filterLogs function.